code/game/atoms.dm

```diff
@@ -642,15 +642,22 @@ var/list/blood_splatter_icons = list()
 	update_icons()	//apply the now updated overlays to the mob
 
 
-/atom/proc/add_vomit_floor(mob/living/carbon/M as mob, var/toxvomit = 0)
-	if( istype(src, /turf/simulated) )
-		var/obj/effect/decal/cleanable/vomit/this = new /obj/effect/decal/cleanable/vomit(src)
+/atom/proc/add_vomit_floor(toxvomit = 0, green = FALSE)
+	playsound(src, 'sound/effects/splat.ogg', 50, 1)
+	if(!isspaceturf(src))
+		var/type = green ? /obj/effect/decal/cleanable/vomit/green : /obj/effect/decal/cleanable/vomit
+		var/vomit_reagent = green ? "green_vomit" : "vomit"
+		for(var/obj/effect/decal/cleanable/vomit/V in get_turf(src))
+			if(V.type == type)
+				V.scoop_reagents[vomit_reagent] += 5

```

Comment:
When I asked what I did, it was more of an actual inquiry than a change this. I'm genuinely curious as to what you're attempting to do here.

I have an idea, but want to be sure.

Reply:
Oh I thought about the if statement. But it's adding puke to the puke pile. So there will still be puke added to the pile in form of more chemicals in the puke decal

Reply:
that's not how `scoop_reagents` works.

It's a list that's adds reagents to the effect on initialization; altering it after runtime will have zero effects.

Reply:
Ah copy. For some reason my intellisense didn't give me the option of reagents so I thought I couldn't use that.